### PR TITLE
Add GET member by alias

### PR DESCRIPTION
**Pull request by danielcooperxyz**
**Problem**
We want to query for a members address by passing in the alias.

**Solution**
Provide an endpoint that will return the address for a given alias.

### CODE REVIEW

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]
Mostly looking good. Afew things:
1. integration tests
2. how can we have a route for getting member by `address`. We will need this. Likely it will mean defining exclusive patterns for `Address` and `Alias`

**Review by dblane-digicatapult — COMMENTED** [review]
Add valid ingress paths to the helm `values.yaml`

File: app/api-v1/routes/members/{aliasOrAddress}.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    parameters: [
+      {
+        description: 'Alias or address of the member',
+        in: 'path',
+        required: true,
+        name: 'aliasOrAddress',
+        allowEmptyValue: true,
+        schema: {
+          // oneOf: [{ $ref: '#/components/schemas/Alias' }, { $ref: '#/components/schemas/Address' }],
+          $ref: '#/components/schemas/Address',
```
Comment (mattdean-digicatapult) [inline app/api-v1/routes/members/{aliasOrAddress}.js:46]: How can an alias get through if we have the stricter `Address` type here in validation? I'm surprised tests pass to say the least

File: test/integration/routes.test.js [hunk 2]
```diff
@@ -116,5 +116,37 @@ describe('routes', function () {
...
+
+      const res = await getMembersByAliasOrAddressRoute(
+        app,
+        '5GNJqTPyNqANBkUVMN1LPPrxXnFouWXoe2wNSmmEoLctxiZY',
+        authToken
+      )
+
+      expect(res.status).to.equal(200)
+      expect(res.body).deep.equal(expectedResult)
+    })
```
Comment (mattdean-digicatapult) [inline test/integration/routes.test.js:150]: These tests rely on previous tests completeing successfully. When you merge in `main` these tests will likely break so you need to do that and then update these

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]

File: app/api-v1/routes/members/{aliasOrAddress}.js [hunk 3]
```diff
@@ -16,9 +16,11 @@ module.exports = function (apiService) {
 
       if (addrRegex.test(aliasOrAddress)) {
         result = await apiService.getMembersByAddress(aliasOrAddress)
-      }
-      if (aliasRegex.test(aliasOrAddress)) {
+      } else if (aliasRegex.test(aliasOrAddress)) {
         result = await apiService.getMembersByAlias(aliasOrAddress)
+      } else {
+        res.status(400)
+        return
```
Comment (mattdean-digicatapult) [inline app/api-v1/routes/members/{aliasOrAddress}.js:23]: Is there a test case you can produce that hit's this? Perhaps a string longer than 255 characters?

Comment (danielcooperxyz) [reply app/api-v1/routes/members/{aliasOrAddress}.js:23]: I have added a test locally that does not fail with a `>255` length string.

Comment (mattdean-digicatapult) [reply app/api-v1/routes/members/{aliasOrAddress}.js:23]: So we need to make that fail somehow. Perhaps the `Alias` regex should be `?!^[1-9A-HJ-NP-Za-km-z]{48}$)^.{0,255}$` instead? This seems to work:

```js
/(?!^[1-9A-HJ-NP-Za-km-z]{48}$)^.{0,255}$/.test(Array(256).fill('a').join('')) // false
/(?!^[1-9A-HJ-NP-Za-km-z]{48}$)^.{0,255}$/.test(Array(256).fill('a').join('')) // true
```

**Review by mattdean-digicatapult — APPROVED** [review]
I'm approving this, but I should admit that I've added some changes on top of those by @danielcooperxyz so I'll need someone else to take a look as well. One shortcoming I'll admin to is that the error handling isn't graceful. I'd like to follow the same approach as we did in `inteli-api` wit hthe error throws, but that's not for this PR. If people would like I'll create a ticket in the backlog

**Review by jonmattgray — APPROVED** [review]

**Comment by jonmattgray** [comment]
> I'm approving this, but I should admit that I've added some changes on top of those by @danielcooperxyz so I'll need someone else to take a look as well. One shortcoming I'll admin to is that the error handling isn't graceful. I'd like to follow the same approach as we did in `inteli-api` wit hthe error throws, but that's not for this PR. If people would like I'll create a ticket in the backlog

https://digicatapult.atlassian.net/browse/IN-251

**Pull request merged**